src/main/java/duke/TaskListStorage.java

```diff
@@ -0,0 +1,258 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.exceptions.IncorrectCommandFormatException;
+import duke.exceptions.InvalidIndexException;
+import duke.exceptions.MissingDescriptionException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+/**
+ * The TaskListStorage class stores an arraylist of tasks.
+ * It contains functionality to manipulate this list (ie, marking, unmarking,
+ * adding, deleting tasks)
+ * It also handles saving/loading the list of tasks to/from a file.
+ *
+ * Notes: Combining Task and Storage is a solution adopted from the discussion
+ * here https://github.com/nus-cs2103-AY2324S1/forum/issues/30
+ */
+public class TaskListStorage {
+    /**
+     * The list of tasks.
+     */
+    private final ArrayList<Task> taskList = new ArrayList<>();
+
+    /**
+     * The file object to save/read the list of tasks to/from.
+     */
+    private final File file;
+
+    /**
+     * The filepath of the file to save the list of tasks to.
+     */
+    private String filepath;
+
+    /**
+     * Creates a TaskListStorage object.
+     * This also loads the list of tasks from the file.
+     *
+     * @param filepath The filepath of the file to save the list of tasks to.
+     */
+    public TaskListStorage(String filepath) {
+        this.filepath = filepath;
+        this.file = new File(filepath);
+        File parentDir = file.getParentFile();
+        if (!parentDir.exists()) {
+            parentDir.mkdirs();
+        }
+
+        if (!this.file.exists()) {
+            try {
+                this.file.createNewFile();
+            } catch (java.io.IOException e) {
+                String errorString = "Error creating file! Make sure you have the correct permissions.\n"
+                        + "Otherwise tasks will not be saved.\n"
+                        + e.getMessage();
+                Ui.printInLine(errorString);
+            }
+        }
+
+        try {
+            this.loadFromFile();
+        } catch (FileNotFoundException | IncorrectCommandFormatException | MissingDescriptionException e) {
+            Ui.printInLine(e.getMessage());
+        }
+    }
+
+    private void loadFromFile()
+            throws FileNotFoundException, IncorrectCommandFormatException, MissingDescriptionException {
+        Scanner sc = new Scanner(this.file);
+        while (sc.hasNextLine()) {
+            String line = sc.nextLine();
+            String[] taskInfo = line.split(" \\| ");
+            String taskType = taskInfo[0];
+            boolean isDone = taskInfo[1].equals("1");
+            String taskDescription = taskInfo[2].replace("\\|", "|"); //Unescape
+            try {
+                switch (taskType) {
```

Comment:
Is there a reason why you did not use an enum for the task types?

Reply:
This is the rawest form of parsing the inputs from the file, as such enums are not necessary. Creating an enum will mean I must then associate the enum with the letter I'm using to save to the file which seems like extra space and computational work for not much increase in understanding/readability of the code. 

Without enums, it should be easy to see D corresponds to creating a deadlineTask since the following line is adding a deadline task to the tasklist.

The only time I might consider using enums is if the task types are a considerably high number that 

1)I cant just use simple alphabets to represent
2)I might need to map what I write in the task file to its own unique task, ie D1, D2 etc.